source/agora/test/Network.d

```diff
@@ -86,3 +86,19 @@ unittest
     auto network = makeTestNetwork!Network(NetworkTopology.Simple, 4);
     network.waitUntilConnected(4);
 }
+
+///
+unittest
+{
+    import std.digest.sha; 
+    auto network = makeTestNetwork!Network(NetworkTopology.Simple, 4);
+    network.waitUntilConnected(4);
+
+    assert (network.apis.keys.length == 4);
+
+    Hash h1 = sha256Of("Message No. 1");
+    auto key = network.apis.keys[0];
```

Comment:
Since `apis` is a hash / associative array, the `keys` you get are not sorted. So you could just use `network.apis.values[0]` and get the same random node.

Reply:
@Geod24,  Move to source/agora/test/GossipProtocol.d
Check Please ~